test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/HtmlHelperLabelExtensionsTest.cs

```diff
@@ -252,6 +252,38 @@ public void LabelFor_DisplaysVariableName()
             Assert.Equal("<label for=\"HtmlEncode[[unknownKey]]\">HtmlEncode[[unknownKey]]</label>", HtmlContentUtilities.HtmlContentToString(result));
         }
 
+        [Fact]
+        public void LabelHelpers_UsesSpecifiedLabelText()
+        {
+            // Arrange
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper();
+
+            // Act
+            var labelResult = helper.Label("Property1", labelText: "Hello");
+            var labelForResult = helper.LabelFor(m => m.Property1, labelText: "Hello");
+            var labelForModelResult = helper.LabelForModel(labelText: "Hello");
```

Comment:
Also this last one is testing something different. Other calls get `<label>` for the `Property1` method but this gets the `<label>` for the entire model. The second is a rather special case since, in many cases, the top-level model cannot be used in HTML helpers.